File: 1/2020.09.22/94.Binary Tree Inorder Traversal/java/solution.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        traversal(root, res);
        return res;
    }

    private void traversal(TreeNode root, List<Integer> res) {
        if (root != null) {
            traversal(root.left, res);
            res.add(root.val);
            traversal(root.right, res);
        }
    }
}

File: 1/2020.09.22/94.Binary Tree Inorder Traversal/python/solution.py
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        self.traversal(root, res)
        return res
    
    def traversal(self, root, result):
        if root:
            self.traversal(root.left, result)
            result.append(root.val)
            self.traversal(root.right, result)
